graylog2-web-interface/src/components/common/Interactable.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Rnd } from 'react-rnd';
+import styled from 'styled-components';
+import { lighten } from 'polished';
+
+import { Button } from 'components/graylog';
+import Icon from 'components/common/Icon';
+import teinte from 'theme/teinte';
+
+const DEFAULT_SIZE = { width: 450, height: 300 };
+const halfWidth = Math.ceil((document.body.offsetWidth / 2) - (DEFAULT_SIZE.width / 2));
+const halfHeight = Math.ceil((document.body.offsetHeight / 2) - (DEFAULT_SIZE.height / 2));
+const stayOnScreenHeight = halfHeight < 0 ? 55 : halfHeight;
+const DEFAULT_POSITION = {
+  x: halfWidth,
+  y: stayOnScreenHeight,
+};
+
+const InteractableWrapper = styled.div`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 1030;
+  pointer-events: none;
+`;
+
+const StyledRnd = styled(Rnd)`
+  box-shadow: 0 0 9px rgba(31, 31, 31, .25),
+              0 0 6px rgba(31, 31, 31, .25),
+              0 0 3px rgba(31, 31, 31, .25);
+  background-color: ${lighten(0.15, teinte.primary.tre)};
+  border-radius: 3px;
+  flex-direction: column;
+  display: flex !important;
+  pointer-events: auto;
+`;
+
+const Content = styled.div`
+  flex: 1;
+  padding: 0 15px;
+`;
+
+const Header = styled.header`
+  padding: 6px 12px 9px;
+  display: flex;
+  align-items: center;
+  background-color: ${lighten(0.25, teinte.primary.tre)};
+  border-bottom: 1px solid ${teinte.primary.tre};
+  border-top-left-radius: 3px;
+  border-top-right-radius: 3px;
+  cursor: move;
+`;
+
+const Title = styled.h3`
+  color: ${teinte.primary.due};
+  flex: 1;
+`;
+
+const DragBars = styled(Icon)`
+  color: ${teinte.secondary.tre};
+`;
+
+/**
+ * A resizable and draggable modal component
+ *
+ * Can be controlled or uncontrolled, using [`react-rnd`](https://github.com/bokuweb/react-rnd) under the hood
+ */
+
+const Interactable = ({
```

Comment:
When I first read the component name I was unsure what the component did, maybe renaming it to `InteractableModal` makes it a bit easier?